main.go

```diff
@@ -453,6 +462,23 @@ func (a *arg) list(proxy string) error {
 	return nil
 }
 
+// buildEnv builds the correct environment for running go commands from gobin.
+// proxy is expected to be empty or take the form "GOPROXY=X". If it is non
+// empty it will be added to the environment.
+func buildEnv(proxy string) []string {
+	env := append(os.Environ(), "GO111MODULE=on")
+	if proxy != "" {
+		env = append(env, proxy)
+	}
+	goflags := os.Getenv("GOFLAGS")
+	if *fMainMod {
+		if *fMod != "" {
+			goflags = strings.Join([]string{goflags, "-mod=" + *fMod}, " ")
```

Comment:
or

    goflags += " -mod=" + *fMod

?

Reply:
Resolving via a different means, but same comment as above

Reply:
Actually, what am I talking about. It's clearly late :) I took your suggestion